pyTransition/transition.py

```diff
@@ -0,0 +1,273 @@
+            departure_or_arrival_choice (string): Specifies whether the used time is "Departure" or "Arrival". Possible values are "Departure" or "Arrival".
+            departure_or_arrival_time (time): Departure or arrival time of the trip
+            max_travel_time_minutes (int): Maximum travel time including access, in minutes
+            min_waiting_time_minutes (int): Minimum waiting time, in minutes
+            max_transfer_time_minutes (int): Maximum transfer time, in minutes
+            max_access_time_minutes (int): Maximum access time, in minutes
+            max_first_waiting_time_minutes (int): Maximum wait time at first transit stop, in minutes
+            with_geojson (bool): If True, the returned JSON file will contain the "pathsGeojson" key for each mode
+            with_alternatives (bool): Indicates if the calculation must include alternative routes or not
+
+        Returns:
+            JSON: Route for each transit mode in JSON format
+        """
+        departure_or_arrival_time = departure_or_arrival_time.hour * 3600 + departure_or_arrival_time.minute * 60 + departure_or_arrival_time.second
+        departure_time = departure_or_arrival_time if departure_or_arrival_choice == "Departure" else None
+        arrival_time = departure_or_arrival_time if departure_or_arrival_choice == "Arrival" else None
+
+        body = {
+            "routingModes" : modes,
+            "withAlternatives" : with_alternatives,
+            "departureTimeSecondsSinceMidnight" : departure_time,
+            "arrivalTimeSecondsSinceMidnight" : arrival_time,
+            "minWaitingTimeSeconds" : min_waiting_time_minutes * 60, 
+            "maxTransferTravelTimeSeconds" : max_transfer_time_minutes * 60,
+            "maxAccessEgressTravelTimeSeconds" : max_access_time_minutes * 60,
+            "maxFirstWaitingTimeSeconds" : max_first_waiting_time_minutes * 60,
+            "maxTotalTravelTimeSeconds" : max_travel_time_minutes * 60,
+            "scenarioId" : scenario_id,
+            "originGeojson" : {
+                "type": "Feature",
+                "id": 1,
+                "geometry": { "type": "Point", "coordinates": origin }
+            },
+            "destinationGeojson" : {
+                "type": "Feature",
+                "id": 1,
+                "geometry": { "type": "Point", "coordinates": destination }
+            }
+        }
+
+        headers = self.__build_headers()
+        params = {"withGeojson": "true" if with_geojson else "false"}
+        response = requests.post(f"{self.base_url}/api/route", headers=headers, json=body, params=params)
+        response.raise_for_status()
+        return response.json()
+
+    def request_accessibility_map(self,
+                                  coordinates,
+                                  scenario_id, 
+                                  departure_or_arrival_choice,
+                                  departure_or_arrival_time: time,
```

Comment:
I might be about to learn something new, what does the : time do here??

Reply:
It was for type-hinting, but we will remove for uniformity

Reply:
ok, that's what I thought, but it was weird to only have type hint in one place.